test/test_decorator.py

```diff
@@ -43,6 +41,14 @@ def pop(self, index, default_value=None):
             return default_value
 
 
+def delay_evaluation_wrapper(fn: Callable) -> Callable:
+    def delayed_call(*args, **kwargs):
+        return lambda: fn(*args, **kwargs)
+
+    return delayed_call
+
+
+@delay_evaluation_wrapper
```

Comment:
```suggestion
@functools.partial
```

Reply:
And at that point, as far as I can tell, this doesn't do anything, so why are we adding it?